internal/operands/common-templates/reconcile.go

```diff
@@ -123,6 +134,51 @@ func reconcileEditRole(request *common.Request) (common.ResourceStatus, error) {
 		})
 }
 
+func reconcileOlderTemplates(request *common.Request) ([]common.ReconcileFunc, error) {
+	// Append functions to take ownership of previously deployed templates during an upgrade
+	existingTemplates := &templatev1.TemplateList{}
+	templatesSelector := func() labels.Selector {
+		tplSelector := labels.NewSelector()
+		tplBaseRequirement, err := labels.NewRequirement("template.kubevirt.io/type", selection.Equals, []string{"base"})
+		if err != nil {
+			panic("Failed creating label selector for 'template.kubevirt.io/type=base")
+		}
+
+		// Only fetching older templates  to prevent duplication of API calls
+		tplVersionRequirement, err := labels.NewRequirement("template.kubevirt.io/version", selection.NotEquals, []string{Version})
+		if err != nil {
+			panic("Failed creating label selector for 'template.kubevirt.io/version")
+		}
+
+		tplSelector = tplSelector.Add(*tplBaseRequirement, *tplVersionRequirement)
+
+		return tplSelector
+	}()
+
+	err := request.Client.List(request.Context, existingTemplates, &client.ListOptions{
+		LabelSelector: templatesSelector,
+		Namespace:     request.Instance.Spec.CommonTemplates.Namespace,
+	})
+
+	// There might not be any templates (in case of a fresh deployment), so a NotFound error is accepted
+	if err != nil && !errors.IsNotFound(err) {
+		return nil, err
+	}
+
+	funcs := make([]common.ReconcileFunc, 0, len(existingTemplates.Items))
+	for i := range existingTemplates.Items {
+		template := &existingTemplates.Items[i]
+		funcs = append(funcs, func(*common.Request) (common.ResourceStatus, error) {
+			// An empty update func is sufficient because the generic reconcile
+			// logic would add owner annotations and remove ownerReferences
```

Comment:
Adding owner annotations means that the SSP resource will own the old templates. Should the old templates be removed in `Cleanup()` method? So the meaning of owner annotations is the same for all resources.

Reply:
I'll include this as part of #71